src/experiments.js

```diff
@@ -168,16 +133,40 @@ export function toggleExperiment(win, experimentId, opt_on,
   return on;
 }
 
+/**
+ * Calculate whether the experiment is on or off based off of the
+ * cookieFlag or the global config frequency given.
+ * @param {!Window} win
+ * @return {!Object<string, boolean>}
+ */
+function experimentToggles(win) {
+  if (toggles_) {
+    return toggles_;
+  }
+  toggles_ = Object.create(null);
+
+  // Read the default config of this build.
+  if (win.AMP_CONFIG) {
+    for (const experimentId in win.AMP_CONFIG) {
+      if (win.AMP_CONFIG.hasOwnProperty(experimentId)) {
+        const frequency = win.AMP_CONFIG[experimentId];
```

Comment:
We probably should use `Object.prototype.hasOwnProperty.call`, otherwise I don't think we can guarantee that it has prototype.